docs/rst/microros_documentation/user_api/user_api_qos.rst

```diff
@@ -0,0 +1,67 @@
+.. _micro_user_api_qos:
+
+QoS configuration
+=================
+
+.. contents:: Table of Contents
+    :depth: 2
+    :local:
+    :backlinks: none
+
+QoS (Quality of Services) configuration related to DDS communication can also applied to micro-ROS entities, as its default middleware `eProsima Micro XRCE-DDS <https://micro-xrce-dds.docs.eprosima.com/en/latest/>`_ allows for complete customization using the provided RMW API.
+
+Further information about ROS 2 available QoS can be found on :ref:`About Quality of Service settings <concepts_about_qos>`.
+
+Initialization
+^^^^^^^^^^^^^^
+
+As shown in the initialization API of each entity, a ``rmw_qos_profile_t`` can be used to configure the QoS of a micro-ROS entity:
+
+.. code-block:: c
+
+  // Create a publisher with depth 100.
+  rcl_publisher_t custom_publisher;
+  rmw_qos_profile_t custom_qos = rmw_qos_profile_default;
+  custom_qos.depth = 100;
+  rclc_publisher_init(
+      &custom_publisher, &node,
+      &type_support, &topic_name, &rmw_qos_profile_default);
+
+This ``rmw_qos_profile_t`` structure is provided by ROS 2 RMW headers: `rmw_qos_profile_t <http://docs.ros2.org/latest/api/rmw/structrmw__qos__profile__t.html>`_. A list with predefined qos profiles is also available: `qos_profiles.h <https://github.com/ros2/rmw/blob/humble/rmw/include/rmw/qos_profiles.h>`__.
+
+.. note::
+
+  By default, the following qos configuration is applied on each entity:
+
+    - Default Publisher/subscriber: `rmw_qos_profile_default <https://github.com/ros2/rmw/blob/humble/rmw/include/rmw/qos_profiles.h#L51>`__.
+    - Best effort Publisher/subscriber : `rmw_qos_profile_sensor_data <https://github.com/ros2/rmw/blob/humble/rmw/include/rmw/qos_profiles.h#L25>`__.
+    - Default Service/client: `rmw_qos_profile_services_default <https://github.com/ros2/rmw/blob/humble/rmw/include/rmw/qos_profiles./h#L64>`__.
+    - Best effort Service/client: `rmw_qos_profile_services_default <https://github.com/ros2/rmw/blob/humble/rmw/include/rmw/qos_profiles.h#L64>`__ with ``reliability = RMW_QOS_POLICY_RELIABILITY_BEST_EFFORT``
+
+Behaviour
+^^^^^^^^^
+
+When micro-ROS history slots are complete and new data arrives, the behaviour is determined by the configured ``history`` kind and ``depth``.
```

Comment:
Make a reference here to the place where we explain the RMW history slots

Reply:
https://github.com/eProsima/vulcanexus/pull/114/commits/1c99fa6141e145405199590760f3c010f12383cd